### PR TITLE
fix gathering helpers

### PR DESCRIPTION
**Pull request by tylerturdenpants**
Naive attempt to get helpers gathered.  As it sits, this works.  I have excluded templates from being invoked.

### CODE REVIEW

File: lib/gather/gather-telemetry.js [hunk 1]
```diff
@@ -56,7 +60,6 @@ module.exports = async function gatherTelemetry(url, puppeteerArgs = {}) {
         'Route',
         'Component',
         'Service',
-        'Helper',
```
Comment (NullVoxPopuli) [inline lib/gather/gather-telemetry.js]: how are helpers identified?

Comment (tylerturdenpants) [reply lib/gather/gather-telemetry.js]: I did learn is there is no Ember Type` Helper` like there is Component and Route.  The only thing I found the allows me to determine a helper is to use `isHelperFactory ` property

**Comment by NullVoxPopuli** [comment]
thanks for investigating this! <3

any chance you'd be willing to open a PR using this gitref as the version for telemetry-helpers and add an additional smoke test in the root tests folder? https://github.com/ember-codemods/ember-no-implicit-this-codemod/tree/master/test

when I ran in to this problem, I had a mix of classic and new stuff -- might have also been due to ember > 3.12.
The existing smoke test app is 3.10 -- which is good, but I think we also need one for 3.12/3.13

**Comment by tylerturdenpants** [comment]
Sure.  I'll see if I have some more time this evening.  I might be to help tomorrow.  I still want to now why this change tho so that I can be more educated about it

File: lib/gather/gather-telemetry.js (hunk 1 above)
Comment (NullVoxPopuli) [reply lib/gather/gather-telemetry.js]: huh, interesting!

**Comment by NullVoxPopuli** [comment]
My hunch is that it has to do with the new property-tracking system introduced in 3.13 -- haven't validated that yet though

**Comment by tylerturdenpants** [comment]
Let me blow your mind:  I didn't see `Helper` when ember-source was 3.8!

Think the long term is to have a smoke test the does the equivalent of ember-try to swap out `ember-source`

**Comment by NullVoxPopuli** [comment]
yeah, I mean that could be done pretty easy, I think. tests/test.js could swap the versions, and run the tests for each version. Like,

`const EMBER_VERSION = process.env.EMBER_VERSION`
// use `fs` to write, or execute a `jq` script to swap it out
run the current code in test.js, extracted to a function.
and then we'd need to set up CI for different EMBER_VERSIONS

**Comment by patocallaghan** [comment]
Good find @tylerturdenpants. I just came across this today running the no-implicit-this codemod against an Ember 3.13 app

**Comment by patocallaghan** [comment]
Not sure if useful but in the `no-curly-component-invocation` lint rule [we advise to do the following](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-curly-component-invocation.md#whitelisting-helpers) in your app to generate a list of helpers.

``` js
var componentLikeHelpers = Object.keys(require.entries)
  .filter(name => name.includes('/helpers/'))
  .map(name => {
    let path = name.split('/helpers/');
    return path.pop();
  })
  .filter(name => !name.includes('/'))
  .uniq();
```

**Comment by patocallaghan** [comment]
@tylerturdenpants @NullVoxPopuli any status update here? We'd love to  use the no-implicit-this-codemod but we're blocked until a fix for helpers is shipped (we're on 3.13). Technically we could fork this ourselves but if there's something can be done to get it upstreamed I'd be happy to help.

Edit: nevermind I just found https://github.com/ember-codemods/ember-codemods-telemetry-helpers/issues/13 which appears to be a more fundamental change

**Comment by NullVoxPopuli** [comment]
I'm writing some tests now to verify that this fixes the issue :+1:

**Comment by NullVoxPopuli** [comment]
Based on: https://github.com/ember-codemods/ember-no-implicit-this-codemod/pull/29
It looks like this works.

anyone else want to review? @pzuraq ?

**Review by NullVoxPopuli — APPROVED** [review]

**Review by rwjblue — CHANGES_REQUESTED** [review]
I think we should consider this blocked on #14

File: lib/gather/analyze-ember-object.js [hunk 2]
```diff
@@ -2,14 +2,27 @@ module.exports = function analyzeEmberObject(possibleEmberObject) {
...
+    (typeof possibleEmberObject.default.proto !== 'function' ||
+      typeof possibleEmberObject.default.create !== 'function')
+  ) {
     return undefined;
   }
-  let proto = possibleEmberObject.default.proto();
+
+  let obj = possibleEmberObject.default.proto
+    ? possibleEmberObject.default.proto()
+    : possibleEmberObject.default.create();
```
Comment (rwjblue) [inline lib/gather/analyze-ember-object.js]: I don't understand this change, why would we `.create()`?

Comment (tylerturdenpants) [reply lib/gather/analyze-ember-object.js]: When investing this bug originally I was able to enumerate (or at least I thought I was) more types of objects when using create if proto was not available. If this is not needed I’m still left trying to answer the original question - how do you tell class based helpers from the old(?) style?  Is isHelperFactory a reliable way?

Comment (rwjblue) [reply lib/gather/analyze-ember-object.js]: > Is isHelperFactory a reliable way?

Yes, that is what I think we should use. I don't think we want to `.create()` things (it is pretty likely to actually cause errors if we `.create()` a thing without the right arguments and stuff).

Comment (tylerturdenpants) [reply lib/gather/analyze-ember-object.js]: great, i will revert a majority of this and push

**Review by rwjblue — APPROVED** [review]

**Pull request merged**